cmake/modules/FindP4EST.cmake

```diff
@@ -149,6 +149,25 @@ IF(EXISTS ${P4EST_INCLUDE_DIR}/p4est_config.h)
     "^[0-9]+\\.[0-9]+\\.[0-9]+\\.([0-9]+).*$" "\\1"
     P4EST_VERSION_PATCH "${P4EST_VERSION}")
 
+  #
+  # Check if the p4est library found is compatible with
+  # the current version of deal.II by comparing it to
+  # a manually set minimum value.
+  #
+  SET(P4EST_VERSION_REQUIRED 2.0)
+
+  IF(P4EST_VERSION LESS P4EST_VERSION_REQUIRED)
+    MESSAGE(FATAL_ERROR "\n"
+      "Deal.II has been configured with DEAL_II_WITH_P4EST="
+      "${DEAL_II_WITH_P4EST}, but no compatible version of p4est "
+      "has been found.\n"
+      "This version of deal.II needs to be configured with "
+      "at least p4est ${P4EST_VERSION_REQUIRED}.\n"
+      "Instead, p4est ${P4EST_VERSION} was found in the folder "
+      "${P4EST_DIR}.\n"
+      )
+  ENDIF()
```

Comment:
(Independent of the discussion whether we want to bump the minimal p4est version)

Please remove this (lines 152 to 169) and simply increase the version number in `configure_p4est.cmake`:
```
    #
    # We require at least version 0.3.4.1
    #
    IF(P4EST_VERSION VERSION_LESS  "0.3.4.1")
      MESSAGE(STATUS "Insufficient p4est installation found: "
        "At least version 0.3.4.1 is required."
        )
      SET(P4EST_ADDITIONAL_ERROR_STRING
        "Insufficient p4est installation found!\n"
        "At least version 0.3.4.1 is required.\n"
        )
      SET(${var} FALSE)
    ENDIF()
```

Reply:
(if you want, feel free to introduce a local variable `SET(_version_required 2.0)` in order to avoid typing the same number multiple types.)